File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        long rev=0;
        while(x!=0){
            int rem=x%10;
            x/=10;
            rev=(rev*10)+rem;
        }
        if(rev>Integer.MAX_VALUE || rev<Integer.MIN_VALUE)
        return 0;
        return (int)rev;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let rev = 0;

    while (x !== 0) {
        let digit = x % 10;
        x = (x / 10) | 0;

        rev = rev * 10 + digit;

        if (rev < -(2 ** 31) || rev > 2 ** 31 - 1) {
            return 0;
        }
    }

    return rev;
};